app/src/main/java/com/gruppo4/audioUtility/AudioUtilityManager.java

```diff
@@ -0,0 +1,173 @@
+    public static final int MIN_PERCENTAGE = 0;
+
+    private static final int VIBRATION_TIME = 500;
+    private static final int VIBRATION_DELAY = 500;
+    private static final int VIBRATION_REPEAT_CODE = 1;
+
+    public enum AUMStream {
+        ALARM,
+        RING,
+        MUSIC
+    }
+
+    /**
+     * @param context The current Context.
+     * @return The current AudioManager instance.
+     */
+    protected static AudioManager getAudioManager(Context context) {
+        return (AudioManager) context.getSystemService(AUDIO_SERVICE);
+    }
+
+    /**
+     * @param context The current Context.
+     * @param stream  The chosen stream (should be ALARM, RING or MUSIC).
+     * @return The current stream Volume (in percentage).
+     */
+    public static int getVolume(Context context, AUMStream stream) {
+        int currentVolume = getAudioManager(context).getStreamVolume(getStream(stream));
+
+        int maxVolume = getMaxVolume(context, stream);
+        return Math.round(MAX_PERCENTAGE * currentVolume / maxVolume);
+    }
+
+    /**
+     * @param context The current Context.
+     * @param stream  The chosen stream (should be ALARM, RING or MUSIC).
+     * @return The maximum stream volume (real value).
+     */
+    private static int getMaxVolume(Context context, AUMStream stream) {
+        return getAudioManager(context).getStreamMaxVolume(getStream(stream));
+    }
+
+    /**
+     * Sets up the stream Volume, given a certain percentage.
+     *
+     * @param context    The current Context.
+     * @param stream     The chosen stream (should be ALARM, RING or MUSIC).
+     * @param percentage Target volume (expressed in %).
+     * @throws IllegalArgumentException if percentage is not between 0 and 100.
+     */
+    public static void setVolume(Context context, AUMStream stream, int percentage) throws IllegalArgumentException {
+        if (percentage < MIN_PERCENTAGE)
+            throw new IllegalArgumentException(" Your value is too low. Please insert a value between 0 and 100.");
+        else if (percentage > MAX_PERCENTAGE) {
+            throw new IllegalArgumentException(" Your value is too high. Please insert a value between 0 and 100.");
+        }
+        int maxVolume = getMaxVolume(context, stream);
+
+        // Calculate the real value of the new volume
+        int newVolume = maxVolume * percentage;
+        newVolume = Math.round(newVolume / MAX_PERCENTAGE);
+
+        // Sets the Volume.
+        getAudioManager(context).setStreamVolume(
+                getStream(stream),
+                newVolume,
+                AudioManager.FLAG_SHOW_UI
+        );
+
+
+    }
+
+    /**
+     * Sets up the stream Volume to its maximum value (in percentage).
+     *
+     * @param context The current Context.
+     * @param stream  The chosen stream (should be ALARM, RING or MUSIC).
+     */
+    public static void setMaxVolume(Context context, AUMStream stream) {
```

Comment:
Method name could be misinterpreted as "set the maxVolume field's value". I suggest something like "maximizeVolume", also I find it slightly redundant since the MAX_PERCENTAGE constant is public, I'd make one of the two private to slim down the public part of the app, not an error though.

Reply:
I personally hadn't thought about that. I just knew there were methods like "getStreamMinVolume" and "getStreamMaxVolume", and "setStreamVolume" in AudioManager , so I thought about something similar.
How about setAsMaxVolume(Context,AUMStream) and setAsMinVolume(Context,AUMStream) ?

Reply:
If "As" stands for Audio Stream the clarity doesn't improve much in my opinion, sounds like "set the max volume for the Audio Stream". You could just keep them like this but I'd go for something like the aforementioned maximizeVolume or maybe even setVolumeToMax, or also just add an Enum and leave only setVolume, the choice is yours though.

Reply:
I meant exactly that, not "AudioStream". Thank you. setVolumeToMin and setVolumeToMax sound way better.